src/static-layout.js

```diff
@@ -173,6 +173,114 @@ export function applyStaticLayout(element, fixIeIntrinsic = false) {
+      sizer.setAttribute('slot', 'i-amphtml-svc');
+      setStyles(sizer, {
+        paddingTop:
+          (getLengthNumeral(height) / getLengthNumeral(width)) * 100 + '%',
+      });
+      element.insertBefore(sizer, element.firstChild);
+      element.sizerElement = sizer;
+    }
+  } else if (layout == Layout.INTRINSIC) {
+    // Intrinsic uses an svg inside the sizer element rather than the padding
+    // trick Note a naked svg won't work becasue other thing expect the
+    // i-amphtml-sizer element
+    const sizer = htmlFor(element)`
+      <i-amphtml-sizer class="i-amphtml-sizer" slot="i-amphtml-svc">
+        <img alt="" role="presentation" aria-hidden="true"
+             class="i-amphtml-intrinsic-sizer" />
+      </i-amphtml-sizer>`;
+    const intrinsicSizer = sizer.firstElementChild;
+    intrinsicSizer.setAttribute(
+      'src',
+      !IS_ESM && fixIeIntrinsic && element.ownerDocument
+        ? transparentPng(
+            element.ownerDocument,
+            devAssertNumber(getLengthNumeral(width)),
+            devAssertNumber(getLengthNumeral(height))
+          )
+        : `data:image/svg+xml;charset=utf-8,<svg height="${height}" width="${width}" xmlns="http://www.w3.org/2000/svg" version="1.1"/>`
+    );
+    element.insertBefore(sizer, element.firstChild);
+    element.sizerElement = sizer;
+  } else if (layout == Layout.FILL) {
+    // Do nothing.
+  } else if (layout == Layout.CONTAINER) {
+    // Do nothing. Elements themselves will check whether the supplied
+    // layout value is acceptable. In particular container is only OK
+    // sometimes.
+  } else if (layout == Layout.FLEX_ITEM) {
+    // Set height and width to a flex item if they exist.
+    // The size set to a flex item could be overridden by `display: flex` later.
+    if (width) {
+      setStyle(element, 'width', width);
+    }
+    if (height) {
+      setStyle(element, 'height', height);
+    }
+  } else if (layout == Layout.FLUID) {
+    element.classList.add('i-amphtml-layout-awaiting-size');
+    if (width) {
+      setStyle(element, 'width', width);
+    }
+    setStyle(element, 'height', 0);
+  }
+  // Mark the element as having completed static layout, in case it is cloned
+  // in the future.
+  element.setAttribute('i-amphtml-layout', layout);
+  return layout;
+}
+
+/**
+ * @typedef {
+ *   {layout: !Layout, height: number, width: number} | {layout: !Layout}
+ * } EffectiveLayout
+ */
+
+/**
+ * Gets the effective layout for an element.
+ * If i-amphtml-layout is available, then directly use that value.
+ * Else calculate layout based on element attributes and return the width/height.
+ *
+ * @param {!Element} element
+ * @return {EffectiveLayout}
+ */
+export function getEffectiveLayout(element) {
+  // Return the pre-existing value if layout has already been applied.
+  const completedLayoutAttr = element.getAttribute('i-amphtml-layout');
+  if (completedLayoutAttr) {
+    return {layout: parseLayout(completedLayoutAttr)};
```

Comment:
This misses the `width` and `height` attributes. Are they important?

If it's only used internally in this file, we could export a wrapper function that just returns the `layout` property and keep this function internal.

Reply:
> This misses the width and height attributes. Are they important?

Only within this module. We externally never need to expose it.
Good thinking re. wrapper func.